### ISSUE TITLE
Improper Handling of references in the API reference

### ISSUE THREAD

**Issue by pradyunsg** [pradyunsg/furo#142]
Reproduction steps:

- Checkout https://github.com/achamayou/CCF/tree/furo_theme
- Run "./livehtml.sh" on an Ubuntu 18.04 machine or similar.
- Then go to "Build Apps" and "Developer API" for example, or any of the pages whose title include "API", or even search for the names of the calls referenced in the screenshots.

If you have trouble getting this to work, I can also spin up a hosted version temporarily.

_Originally posted by @achamayou in https://github.com/pradyunsg/furo/discussions/137#discussioncomment-664275_

**Comment by achamayou** [comment]
@jrast only internal references are set to `float: right`, see https://github.com/pradyunsg/furo/commit/bb46d32aa4a977a0c825ec14b82fbfd04be4f0a4

I believe the intention, as suggested in the comment, is to align `[source]` to the right, which is itself fine. But `[source]` isn't the only internal reference, types in function signatures can often be internal references as well.

**Comment by jrast** [comment]
I see. The [source] should have a `float:right`, but return values, which are, in some cases, internal references, should not float right. 

The [source] links seem to be structured like this:

````html
<a class="reference internal" href="...">
    <span class="viewcode-link">
         <span class="pre">[source]</span>
    </span>
</a>
````

while the return type is  structured like this:

````html
<a class="reference internal" href="..." title="...">
    <span class="pre">lib.testprocedure.base.TestOutcome</span>
</a>
````

So basically, based on the class of the inner "span" element one can see which is which. However, in pure CSS it is not possible to apply a style to the parent, given a specific child (at least I'm not aware of a method).

**Comment by achamayou** [comment]
@jrast that's right, I looked for a way to do this and could not find one either. The most you can do is apply to an element if it's empty or not it seems, but the contents can't be used in the selector.

**Comment by jrast** [comment]
A possible solution with JS:

````js
window.addEventListener("load", function() {
    var refs = document.getElementsByClassName('reference internal')
    for(var ref of refs) {
        if(ref.children.length === 0 || !ref.children[0].classList.contains("pre")) {
            continue
        }
        ref.style = "float:none;"
    }
})
````

Note: I have only done a quick check on one of my documentations. But this might provide a starting point for a JS solution.

**Comment by pradyunsg** [comment]
Fixed in #159 thanks to @sirosen! ^>^

**Comment by achamayou** [comment]
Thank you @sirosen and @pradyunsg, the latest release does display all our API doc correctly now, wonderful!
